### PR TITLE
Notebook to use pnid_parser

### PR DESCRIPTION
**Pull request by lynochka**
https://cognitedata.atlassian.net/browse/CONTEXT-1348

The notebook uses experimental SDK.

### CODE REVIEW

File: contextualize_pnid_files.ipynb [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Map asset names to a list with asset_ids with this name\n",
+    "asset_id_dict = dict()\n",
+    "for asset in assets:\n",
+    "    name = asset.name\n",
+    "    if not name in asset_id_dict:\n",
```
Comment (olacognite) [inline contextualize_pnid_files.ipynb]:
```suggestion
    "    if name not in asset_id_dict:\n",
```

File: contextualize_pnid_files.ipynb [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    "    entities_found = [result[\"text\"] for result in results]\n",
+    "    if not entities_found:\n",
+    "        # skip to the next file if no assets are found\n",
+    "        continue\n",
+    "    \n",
+    "    # remove duplicate entities\n",
+    "    entities_found = list(set(entities_found))\n",
+    "    \n",
+    "    # make a list of asset_ids based on the found entities \n",
+    "    asset_ids_found = []\n",
```
Comment (olacognite) [inline contextualize_pnid_files.ipynb]:
```suggestion
    "    asset_ids_found = set()\n",
```

File: contextualize_pnid_files.ipynb [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    "    for entity in entities_found:\n",
+    "        # asset name could correspond to several asset_ids\n",
+    "        asset_ids_found.extend(asset_id_dict[entity])\n",
```
Comment (olacognite) [inline contextualize_pnid_files.ipynb]:
```suggestion
    "        asset_ids_found.update(asset_id_dict[entity])\n",
```

File: contextualize_pnid_files.ipynb [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    "        \n",
+    "    asset_names      = ','.join(map(str, entities_found))\n",
+    "    print(f\"\\tFound assets: {asset_names}\")\n",
+    "    \n",
+    "    # update metadata for the original file\n",
+    "    file_metadata_update = FileMetadataUpdate(id=file.id).asset_ids.set(asset_ids_found)\n",
```
Comment (olacognite) [inline contextualize_pnid_files.ipynb]:
```suggestion
    "    file_metadata_update = FileMetadataUpdate(id=file.id).asset_ids.set(list(asset_ids_found))\n",
```

**Review by olacognite — COMMENTED** [review]
A couple of small suggestions. Don't know how metadata handles duplicate ids, would maybe also be nice with a warning if an asset name with multiple ids is found, but not sure if it is too important

**Comment by lynochka** [comment]
@olacognite great suggestions! @andeplane good to go?

**Review by olacognite — APPROVED** [review]
LGTM

File: contextualize_pnid_files.ipynb [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "project=\"project\"\n",
+    "api_key = os.getenv(\"COGNITE_API_KEY\")\n",
```
Comment (andeplane) [inline contextualize_pnid_files.ipynb]: We don't have api keys in the environment variable here. Can you change so you  import getpass and use 
`api_key = getpass()`?

File: contextualize_pnid_files.ipynb [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    "\n",
+    "client = CogniteClient(api_key=api_key, \n",
```
Comment (andeplane) [inline contextualize_pnid_files.ipynb]:
```suggestion
    "client = CogniteClient(api_key=getpass(), \n",
```
or this

File: contextualize_pnid_files.ipynb [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+    "                       client_name=\"local-jupyter-notebook\", \n",
```
Comment (andeplane) [inline contextualize_pnid_files.ipynb]:
```suggestion
    "                       client_name=\"dshub\", \n",
```

File: contextualize_pnid_files.ipynb [hunk 8]
```diff
@@ -0,0 +1,207 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
```
Comment (andeplane) [inline contextualize_pnid_files.ipynb:9]:
```suggestion
    "import os\n",
    "import getpass\n",
```

File: contextualize_pnid_files.ipynb [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get files with source\n",
+    "source = \"Pearl: ASU 1 Unit 12100 PEFS\"\n",
```
Comment (andeplane) [inline contextualize_pnid_files.ipynb]: Where does this source come from? Can we choose a more intuitive example?

File: contextualize_pnid_files.ipynb [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+    "    # remove duplicate entities\n",
+    "    entities_found = list(set(entities_found))\n",
+    "    \n",
+    "    # make a list of asset_ids based on the found entities \n",
+    "    asset_ids_found = set()\n",
+    "    for entity in entities_found:\n",
+    "        # asset name could correspond to several asset_ids\n",
+    "        asset_ids_found.update(asset_id_dict[entity])\n",
+    "        \n",
+    "    asset_names      = ','.join(map(str, entities_found))\n",
```
Comment (andeplane) [inline contextualize_pnid_files.ipynb]:
```suggestion
    "    asset_names = ','.join(map(str, entities_found))\n",
```

File: contextualize_pnid_files.ipynb [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+    "    file_metadata_updates.append(file_metadata_update)\n",
+    "    \n",
+    "    # add metadata to the new svg, e.g.,\n",
+    "    svg_metadata = {\"OriginalSource\": file.source}\n",
+    "    \n",
+    "    # keep the name for the file, replacing the extention\n",
+    "    name_svg = os.path.splitext(file.name)[0] + \".svg\"\n",
+    "    \n",
+    "    # give a meaningful external_id\n",
+    "    external_id_svg = file.external_id + \"+svg\"\n",
```
Comment (andeplane) [inline contextualize_pnid_files.ipynb:139]: What if the file does not have an external id? Maybe it's ok as this is just a template.

**Review by andeplane — CHANGES_REQUESTED** [review]
Didn't run the notebook, but I assume it works. I have a couple of comments which are mostly just nitpick. The most important one is the api key that in DSHub should come from `getpass`. If you switch that out, and see what you think about the others, good to go!

File: contextualize_pnid_files.ipynb (hunk 11 above)
Comment (lynochka) [reply contextualize_pnid_files.ipynb:139]: I think it's fine with external_id, application teams argument for external_ids so OK in the template. The user of the template could delete it if they want when copying.

Comment (andeplane) [reply contextualize_pnid_files.ipynb:139]: I agree!

**Review by andeplane — APPROVED** [review]
Nice work! LGTM

**Pull request merged**